packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -5017,7 +5017,7 @@ describeWithDOM('mount', () => {
         }
 
         const wrapper = mount(<div><Bar render={() => <div><Foo /></div>} /></div>);
-        expect(() => wrapper.find(Bar).renderProp('render')).to.throw(RangeError);
+        expect(() => wrapper.find(Bar).renderProp('render')()).to.throw(RangeError);
```

Comment:
similarly, i'd expect this to throw immediately, and not inside the returned function.